index.go

```diff
@@ -185,6 +188,25 @@ func (idx *faissIndex) SearchWithoutIDs(x []float32, k int64, exclude []int64, p
 	return
 }
 
+func (idx *faissIndex) SearchWithIDs(x []float32, k int64, include []int64,
+	params json.RawMessage) (distances []float32, labels []int64, err error,
+) {
+	includeSelector, err := NewIDSelectorBatch(include)
+	defer includeSelector.Delete()
```

Comment:
maybe this bit got missed, but NewIDSelectorBatch() does return nil when err != nil so I think this should be moved to line > 198

Reply:
done, thanks!